docs/Course Wiki/CS Course Offerings/CS374A.md

```diff
@@ -0,0 +1,95 @@
+# CS374A
+
+## Content Covered
+
+### First third (languages)
+
+- Induction, with a focus on application to strings
+- Regular languages
+- Regular expressions
+- Deterministic finite autoamta (DFAs)
+- Nondeterminstic finite automata (NFAs)
+- Regular language transformation via DFA and NFA manipulations
+- Myhill-Nerode theorem and proving irregularity with fooling sets
+- Context free languages and context free grammars (CFGs)
+- Turing machines (only some semesters)
+
+### Middle third (algorithms)
+
+- Divide and conquer algorithms
+- Backtracking algorithms
+- Dynamic programming algorithms
+- Graph algorithms
+- Greedy algorithms
+
+### Final third (complexity)
+
+- Undecidable languages
+- Undecidability of the halting problem
+- NP-Completeness and NP-hardness reductions
+
+## Prerequisites
+
+- CS173
+- CS225
+
+CS374A studies algorithms using theoretical tools covered in CS173 and data structures covered in CS225. For the first third, students should be comfortable with induction and finite state machines (FSMs) from CS173; for the second third, students should be comfortable with data structures from CS225; for the final third, students should be comfortable with proof by contradiction from CS173.
+
+## When to Take it
+
+CS374A is a required course for Computer Science and Computer Engineering majors. It serves as a prerequisite for CS421, a required course for Computer Science majors, and students should complete it by their second-to-last semester. Algorithmic topics covered in CS374, especially dynamic programming, are useful for internship and full-time job interviews, so taking the course earlier may be beneficial.
+
+Many students find CS374a to be a challenging course, so use caution when scheduling it in a semester with many other difficult courses.
+
+## Course Structure
```

Comment:
Why is this blank?

Reply:
I think the later sections are part of course structure, so there doesn't need to be anything here.